modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js

```diff
@@ -3816,7 +3815,23 @@ AUI.add(
 							value = [''];
 						}
 					}
-
+					const values = instance.getInputNode().all('option').val();
+					let count = 0;
+					for (let i = 0; i < value.length; i++) {
+						if (!values.includes(value[i])) {
+							count++;
+						}
+					}
+					if (count === value.length) {
+						const name = instance.get('name');
+						const field = instance.getFieldByNameInFieldDefinition(
+							name
+						);
+						value =
+							field.predefinedValue[
+								instance.get('displayLocale')
+							];
+					}
```

Comment:
Hi @diogomelo09, thanks for sending this PR! I think we might not be covering the cases where there are multiple selectable options, multiple predefined values and multiple missing values here.

Please try the following and see if it works better:

```
setValue(values) {
	const instance = this;
	const currentlyAvailableValues = instance
		.getInputNode()
		.all('option')
		.val();
	const predefinedValues = JSON.parse(
		instance.getFieldByNameInFieldDefinition(
			instance.get('name')
		).predefinedValue[instance.get('displayLocale')]
	);

	if (Lang.isString(values)) {
		if (values !== '') {
			values = JSON.parse(values);
		} else {
			values = [''];
		}
	}

	let resultValues = [];

	values.forEach((value) => {
		if (!currentlyAvailableValues.includes(value)) {
			resultValues.push(...predefinedValues);
		} else {
			resultValues.push(value);
		}
	})
	
	resultValues = [...new Set(resultValues)];

	instance
		.getInputNode()
		.all('option')
		.each((item) => {
			item.set(
				'selected',
				resultValues.indexOf(item.val()) > -1
			);
		});
},
```
let me know if you need help :)